homeassistant/components/nws/weather.py

```diff
@@ -0,0 +1,306 @@
+                     'rain_fzra', 'snow_fzra']),
+    ('hail', []),
+    ('lightning-rainy', ['tsra', 'tsra_sct', 'tsra_hi']),
+    ('lightning', []),
+    ('pouring', []),
+    ('rainy', ['rain', 'rain_showers', 'rain_showers_hi']),
+    ('windy-variant', ['wind_bkn', 'wind_ovc']),
+    ('windy', ['wind_skc', 'wind_few', 'wind_sct']),
+    ('fog', ['fog']),
+    ('clear', ['skc']),  # sunny and clear-night
+    ('cloudy', ['bkn', 'ovc']),
+    ('partlycloudy', ['few', 'sct'])
+])
+
+FORECAST_CLASSES = {
+    ATTR_FORECAST_DETAIL_DESCRIPTION: 'detailedForecast',
+    ATTR_FORECAST_TEMP: 'temperature',
+    ATTR_FORECAST_TIME: 'startTime',
+}
+
+FORECAST_MODE = ['daynight', 'hourly']
+
+WIND_DIRECTIONS = ['N', 'NNE', 'NE', 'ENE',
+                   'E', 'ESE', 'SE', 'SSE',
+                   'S', 'SSW', 'SW', 'WSW',
+                   'W', 'WNW', 'NW', 'NNW']
+
+WIND = {name: idx * 360 / 16 for idx, name in enumerate(WIND_DIRECTIONS)}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_MODE, default='daynight'): vol.In(FORECAST_MODE),
+    vol.Optional(CONF_STATION, default=''): cv.string,
+    vol.Required(CONF_API_KEY): cv.string
+})
+
+
+def parse_icon(icon):
+    """
+    Parse icon url to NWS weather codes.
+
+    Example:
+    https://api.weather.gov/icons/land/day/skc/tsra,40/ovc?size=medium
+
+    Example return:
+    ('day', (('skc', 0), ('tsra', 40),))
+    """
+    icon_list = icon.split('/')
+    time = icon_list[5]
+    weather = [i.split('?')[0] for i in icon_list[6:]]
+    code = [w.split(',')[0] for w in weather]
+    chance = [int(w.split(',')[1]) if len(w.split(',')) == 2 else 0
+              for w in weather]
+    return time, tuple(zip(code, chance))
+
+
+def convert_condition(time, weather):
+    """
+    Convert NWS codes to HA condition.
+
+    Choose first condition in CONDITION_CLASSES that exists in weather code.
+    If no match is found, return fitst condition from NWS
+    """
+    conditions = [w[0] for w in weather]
+    prec_prob = [w[1] for w in weather]
+
+    # Choose condition with highest priority.
+    cond = next((key for key, value in CONDITION_CLASSES.items()
+                 if any(condition in value for condition in conditions)),
+                conditions[0])
+
+    if cond == 'clear':
+        if time == 'day':
+            return 'sunny', max(prec_prob)
+        if time == 'night':
+            return 'clear-night', max(prec_prob)
+    return cond, max(prec_prob)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the nws platform."""
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    station = config.get(CONF_STATION)
+    api_key = config.get(CONF_API_KEY)
```

Comment:
Use
```suggestion
    api_key = config[CONF_API_KEY]
```
as this key has to be present.